modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -1274,15 +1224,20 @@ public File getDataDir() {
      */
     public void verifyInstallation() throws ConfigurationException {
         checkJavaVersion();
-        ifNotExistsAndIsDirectoryThenCreate(getLogDir());
-        ifNotExistsAndIsDirectoryThenCreate(getPidDir());
-        ifNotExistsAndIsDirectoryThenCreate(getDataDir());
-        ifNotExistsAndIsDirectoryThenCreate(getTmpDir());
-        ifNotExistsAndIsDirectoryThenCreate(getPackagesDir());
+        createDirectoryIfDoesntExist(getLogDir());
+        createDirectoryIfDoesntExist(getPidDir());
+        createDirectoryIfDoesntExist(getDataDir());
+        createDirectoryIfDoesntExist(getTmpDir());
+        createDirectoryIfDoesntExist(getPackagesDir());
         checkAddressesAndPorts();
         serverConfigurator.verifyInstallation();
         backingServicesConfigurator.verifyInstallation();
+    }
 
+    private void createDirectoryIfDoesntExist(File directory) {
+        if (!directory.isDirectory()) {
+            directory.mkdirs();
```

Comment:
I think you can just call `mkdirs()` on all the dirs wihtout the `isDirectory` check, it shouldn't fail if the directory already exists.